WcaOnRails/app/controllers/api/v0/api_controller.rb

```diff
@@ -128,6 +128,13 @@ def show_user(user)
                                            .map(&:competition)
                                            .select(&:upcoming?)
       end
+      if params[:ongoing_competitions]
+        json[:ongoing_competitions] = user.registrations
+                                          .accepted
+                                          .includes(competition: [:delegates, :organizers, :events])
+                                          .map(&:competition)
+                                          .select(&:ongoing?)
```

Comment:
The entire notion of querying a user's competitions is identical to `upcoming_competitions` above.

```ruby
user.registrations
.accepted
.includes(competition: [:delegates, :organizers, :events])
.map(&:competition)
```

Please _at least_ move that part out of the two if-blocks and only apply the `select` filter based on which query param is present.
Ideally, consider moving that entire block into the `user.rb` model as a kind of `accepted_competitions` method. (although that's arguably a bit out of scope of this PR, so I leave it to you to decide)